velox/exec/Driver.h

```diff
@@ -395,9 +395,9 @@ struct DriverFactory {
   /// exchange.
   std::optional<core::PlanNodeId> needsExchangeClient() const {
     VELOX_CHECK(!planNodes.empty());
-    if (auto exchangeNode = std::dynamic_pointer_cast<const core::ExchangeNode>(
-            planNodes.front())) {
-      return exchangeNode->id();
+    const auto& leafNode = planNodes.front();
```

Comment:
Is it source node?
```
s/leafNode/sourceNode/
```

Reply:
No it is leaf node. Note that here driver puts nodes in a vector with the front being leaf.